sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -344,8 +344,25 @@ public void setEnabled(boolean enabled) {
         } else {
             suspend(true, new CancellationException());
         }
+        notifyListenersThatChannelStateHasChanged(true);
+    }
 
-        /* Notify listeners that channel state has changed. */
+    /**
+     * Disable channel without removing logs.
+     *
+     */
+    @Override
+    public void disableWithoutRemovingLogs() {
+        mEnabled = false;
```

Comment:
@MatkovIvan @AnastasiaKubova I think we should use a separate flag here, to avoid inconsistency in the SDK state.
e.g.
```
AppCenter.setInstanceNetworkRequestsAllowed(false);
AppCenter.start()
```
And if we call 
```
AppCenter.setEnabled(true) 
```
network requests will be turned on implicitly and it is not what users expect.

Reply:
Also,
```
AppCenter.setInstanceNetworkRequestsAllowed(false); 
AppCenter.start()
```
`isEnabled` method will be showing `false`, while actually the SDK will be storing logs without sending them. It is different from the current behavior and is not expected by users.

Reply:
In the first case, the state `isEnabled` of App Center SDK should not change because this flag is configured [here](https://github.com/microsoft/appcenter-sdk-android/pull/1527/files#diff-aedf3d478ca92d7c2fbe38f3bb991feb739e62d4dc9089c1474e7f72cabe556aR867).

But yeah, as we discussed locally the status of App Center SDK, in this case, is not clear. I am going to revert this changed to my previous variant.